client/src/features/products/productItem.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+  Card,
+  CardActions,
+  CardContent,
+  CardMedia,
+  Button,
+  Typography,
+  Box,
+} from "@mui/material";
+import { Product } from "../../app/interfaces";
+import { useNavigate } from "react-router-dom";
+import { useAppDispatch } from "../../app/hooks";
+import { selectProduct } from "./productSlice";
+import {
+  useGetCurrentUserDetailsQuery,
+} from "../users/usersApi";
+import {
+  useAddToCartMutation,
+  useCreateOrderMutation,
+  useGetUserOrderQuery,
+} from "../orders/ordersApi";
+
+export const ProductItem = (product: Product) => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const [createOrderInDB] = useCreateOrderMutation();
+  const [createProductOrder] = useAddToCartMutation();
+  const { data: user, refetch } = useGetCurrentUserDetailsQuery();
+  const { data: order } = useGetUserOrderQuery();
+
+  const addToCart = async () => {
+    console.log(order);
+    await createOrderInDB({ userId: user!.id });
+    await createProductOrder({
+      productId: product.id,
+      orderId: order!.id,
+      price: product.price,
+      quantity: 1,
+    });
+  };
```

Comment:
This function should probably be extracted into a helper since it might be used somewhere else